snapcraft/plugins/_python/_pip.py

```diff
@@ -374,9 +416,18 @@ def env(self):
 
         return env
 
-    def _run(self, args, **kwargs):
+    def _run(self, args, runner=None, **kwargs):
         env = self.env()
 
-        return snapcraft.internal.common.run_output(
+        # Using None as the default value instead of common.run so we can mock
```

Comment:
How about importing `run_output` by name so it can be mocked "normally"? If that's possible I would slightly prefer it, since adding options for the sole purpose of testing seems less desirable.

Reply:
The option here wasn't added for testing, it was added so we didn't have two run functions doing essentially the same thing but with different runners, which would introduce an opportunity to diverge. Normally this function declaration would look like `def _run(self, args, runner=snapcraft.internal.common.run, **kwargs)`, but default arguments are evaluated at import time, not run time, thus it would still call the actual `snapcraft.internal.common.run` even if it was mocked. So I used `None` instead and wrote that comment.

Reply:
Ah. That makes sense. Thanks for clarifying.